internal/services/appservice/linux_function_app_data_source.go

```diff
@@ -28,8 +28,9 @@ type LinuxFunctionAppDataSourceModel struct {
 	ServicePlanId      string `tfschema:"service_plan_id"`
 	StorageAccountName string `tfschema:"storage_account_name"`
 
-	StorageAccountKey string `tfschema:"storage_account_access_key"`
-	StorageUsesMSI    bool   `tfschema:"storage_uses_managed_identity"` // Storage uses MSI not account key
+	StorageAccountKey       string `tfschema:"storage_account_access_key"`
+	StorageUsesMSI          bool   `tfschema:"storage_uses_managed_identity"` // Storage uses MSI not account key
+	StorageKeyVaultSecretID string `tfschema:"storage_key_vault_secret_id"`
```

Comment:
I would propose to rename this argument to `storage_key_vault_reference_string` in order to align with the MS docs on [Key Vault references](https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references#reference-syntax). Note users might want to use the alternative syntax which doesn't use the secret uri. Either way, the argument should be renamed (in all resources) to reflect that more than just the secret id is required.

Reply:
Hi @fvdnabee - Thanks for the suggestion. Since we refer to this in the provider as the [Secret ID](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_secret#id) on the resource we're referencing, within the context of Terraform and the provider the name here is appropriate and consistent.

Reply:
Even MS names it (i.e. the secretURI) the `secret id`, cfr. https://docs.microsoft.com/en-us/rest/api/keyvault/secrets/get-secret/get-secret#secretbundle. So yeah disregard my comment.